src/core/depth_layer.cpp

```diff
@@ -16,24 +16,21 @@
 
 #include "mir/depth_layer.h"
 
-#ifndef __clang__
-#include "mir/fatal.h"
-#endif
+#include <utility>
 
-auto mir::mir_depth_layer_get_index(MirDepthLayer depth_layer) -> unsigned int
+auto mir::mir_depth_layer_get_index(MirDepthLayer const depth_layer) -> unsigned int
 {
     switch (depth_layer)
     {
-    case mir_depth_layer_background:        return 0;
-    case mir_depth_layer_below:             return 1;
-    case mir_depth_layer_application:       return 2;
-    case mir_depth_layer_always_on_top:     return 3;
-    case mir_depth_layer_above:             return 4;
-    case mir_depth_layer_overlay:           return 5;
+    case mir_depth_layer_background:
+    case mir_depth_layer_below:
+    case mir_depth_layer_application:
+    case mir_depth_layer_always_on_top:
+    case mir_depth_layer_above:
+    case mir_depth_layer_overlay:
+        return depth_layer;
+
+    default:
+        std::unreachable();
```

Comment:
Oh. Do you know if clang will fail the build if this `std::unreachable()` is actually reachable? (ie: if we add an extra enumeration value?)